egs/wsj/s5/steps/nnet3/chain/train.py

```diff
@@ -284,6 +284,13 @@ def ProcessArgs(args):
     if args.chunk_right_context < 0:
         raise Exception("--egs.chunk-right-context should be non-negative")
 
+    if not args.left_deriv_truncate is None:
+        args.deriv_truncate_margin = -args.left_deriv_truncate
+        logger.warning("--chain.left-deriv-truncate (deprecated) is set by user, so --trainer.deriv-truncate-margin is set to negative of that value={0}.".format(args.deriv_truncate_margin))
+
+    if (not args.deriv_truncate_margin is None) and args.deriv_truncate_margin < 0:
```

Comment:
Actually, since you're using None for the default, there is no need to specify that it must be >= 0.  You can remove that check.